src/kOS/Suffixed/Part/PartValue.cs

```diff
@@ -161,5 +162,15 @@ public override int GetHashCode()
         {
             return !Equals(left, right);
         }
+
+        public static SuffixedList<PartValue> PartsToList(IEnumerable<global::Part> parts, SharedObjects shared)
```

Comment:
While I like the idea of eventually making it so that we are using a sort of ListValue that enforces that only kOS objects are ever seen by a kerboscript, I don't really think it can work as a halfway solution like this, and trying to do it will just mean more work later if we have to rip it out and re-do it, thus my comment.  I think the path to doing what you're trying for here would be this:

Step 1: Make a base class (or an interface - I'm thinking interface so we can multiply inherit), let's call it KOSUserObject, from which ALL objects kerboscript sees derives from.  When a user does SET X TO 1. They don't get a native Integer, they get a KOSNumber which is a subclass of KOSUserObject.  When a user does SET S TO "HELLO" they don't get a native string, they get a KOSString, which is a subclass of KOSUserObject, and so on.

Step 2: Enforce this in the stack and in the dictionary - making it only accept a limited set of objects - all of type KOSObject.  A KOSObject is either a KOSUserObject or a KOSMachineObject - UserObjects being the stuff script see and MachineObjects being stuff the VM needs to exist on the stack, like delegates and the like.

Step 3: make it so that the LIST() that scripts use is only allowed to contain objects of type KOSUserObject.